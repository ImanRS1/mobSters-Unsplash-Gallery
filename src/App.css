@import url("./styles/normalize.css");
@import url('https://fonts.googleapis.com/css2?family=Lobster&family=Montserrat:wght@500&display=swap');

body {
  width: 100vw;
  min-height: 100vh;
  background-color: #fcf6f5ff;
  font-family: 'Montserrat', sans-serif;
}

h1 {
  font-family: 'Lobster', cursive;
}

html {
  width: 100vw;
  min-height: 100vh;
}

.App {
  position: relative;
  min-height: 100vh;
}
