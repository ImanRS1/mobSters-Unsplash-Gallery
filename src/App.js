import './App.css';
import React, { useState } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import axios from 'axios';
import Header from './components/Header';
import Search from './components/Search';
import ImageBoard from './components/ImageBoard';
import Pagination from './components/Pagination';
import Footer from './components/Footer';
import Welcome from './components/Welcome';
import About from './components/About';

const App = () => {
  const [input, setInput] = useState('');
  const [page, setPage] = useState(0);
  const [totalPages, setTotalPages] = useState(0);
  const [data, setData] = useState([]);

  const updateInput = (searchInput) => {
    setInput(searchInput);
  };
  
  const config = {
    unsplashKey: 'lZ-oVPWLFJv-IRXFSqoTPMeyuQAXJ3IscBdCfiqSeIw',
  };

  const getData = async (input, page) => {
    const data = await axios.get(`https://api.unsplash.com/search/photos/?query=${input}&page=${page}&client_id=${config.unsplashKey}`);
    setTotalPages(data.data.total_pages);
    setData(data.data.results);
  };

  const updatePage = (input, num=1) => {
    setPage(num);
    getData(input, num);
  };

  return (
    <BrowserRouter>
      <div className="App">
        <Header input={input}/>
        <Routes>
          <Route path="/" element={
            <div>
              <Search updateInput={updateInput} input={input} updatePage={updatePage} page={page} />
              <Welcome />
            </div>
          } />
          <Route path="/search" element={
            <Search updateInput={updateInput} input={input} updatePage={updatePage} page={page} />
          }/>
          <Route path="/search/:input" element={
            <div>
              <Search updateInput={updateInput} input={input} updatePage={updatePage} page={page} />
              <ImageBoard data={data} />
              <Pagination page={page} updatePage={updatePage} totalPages={totalPages} input={input} />
            </div>
          } />
          <Route path="/about" element={<About />} />
        </Routes>
        <Footer />
      </div>
    </BrowserRouter>
  );
}

export default App;
