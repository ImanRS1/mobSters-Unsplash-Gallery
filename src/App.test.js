import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';

describe('Basic App test', () => {
  const div = document.createElement('div');

  test('Renders', () => {
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  test('App snapshot', () => {
    const AppComponent = renderer.create(<App />).toJSON();
    expect(AppComponent).toMatchSnapshot();
  });
});


