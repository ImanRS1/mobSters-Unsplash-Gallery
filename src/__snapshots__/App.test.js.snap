// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic App test App snapshot 1`] = `
<div
  className="App"
>
  <header
    className="sc-bdvvtL hHHJBI"
  >
    <h1
      className="sc-gsDKAQ dTOPoC"
    >
      mobSters Gallery
    </h1>
    <nav>
      <ul>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          href="/search/"
          onClick={[Function]}
        >
          Search
        </a>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </ul>
    </nav>
  </header>
  <div>
    <form
      className="sc-dkPtRN RMDLC"
      onSubmit={[Function]}
    >
      <input
        className="search-input"
        onChange={[Function]}
        placeholder="Search..."
        required={true}
        type="text"
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
    <div>
      <h2>
        Welcome to mobSters Gallery
      </h2>
      <h3>
        Let's start searching together
      </h3>
    </div>
  </div>
  <footer
    className="sc-kDTinF iVCBnx"
  >
    <p>
      GitHub Icon
    </p>
  </footer>
</div>
`;
